lib/surface/api.ex

```diff
@@ -395,6 +425,38 @@ defmodule Surface.API do
     [:required, :props, :as]
   end
 
+  defp get_required_opts(:data, :upload, _opts) do
+    [:accept]
+  end
+
+  defp get_required_opts(_, _, _), do: []
+
+  defp validate_opt_ast!(:data, option, args_ast, _caller)
```

Comment:
It works, but I am pretty sure it's not the right way to do :sweat_smile: Moreover, I do not know how to test it.

Reply:
This is not gonna work for private functions. We have to escape this value so it can be evaluated only inside the `mount` scope. I'll run some experiments.